### DISCUSSION TITLE
Code of Conduct

### DISCUSSION THREAD

**Discussion by sramkrishna** [hpc-social/hpc-social.github.io#53] [General]
Do we have a code of conduct that we are using for the HPC community? Safety first.

**Comment by vsoch** [comment]
This is a harder task than appears at face value. Right now the main place for content is mastodon, and we have a set of moderators there, and (I think) the goal was to follow the other server's as closely as possible https://mast.hpc.social/privacy-policy.

The challenge with CoC isn't writing them but enforcing them. I had a terrible experience with a community earlier this year where they totally just weren't prepared for a report, and it had sad consequences for both sides.

**Reply by sramkrishna** [reply]
Right - but the problem is that many won't participate without a code of conduct at least in the communities in. I can possibly help with code of conduct. I have friends who are in code of conduct committees. My wife is a DEI expert. So I have plenty of ways to help here if you want it.

In anyways, I agree that enforcement actions are problematic - but I do have access to people and resources to help mitigate this. But I would put one in place as it would lead to criticism. Especially when HPC can incorporate things like AI and that crowd is quite sensitive these days about equity.

**Reply by vsoch** [reply]
Oh that sounds fantastic! :partying_face: I definitely don't have that experience, @sramkrishna could you lead the effort? Let us know what you need? Pinging @alansill

**Reply by sramkrishna** [reply]
Sure, how do you want me to help? I never had to pick a code of conduct. But generally, I think there are code of conducts that one can already use. I'll do some research.

**Reply by alansill** [reply]
I do think the only thing we have tht would be covered by a code of conduct on hpc.social right now is the Mastodon instance, and the policies we have in place are designed to promote civil conversation while hitting the required legal points for hosting a server in the US as well as the upstrem requiremenst of our hosting provider. 

To go further, I think the starting point would be the results of the survey we conducted, and we can certainly do others. For now, search on "code" in the results link at [https://survey.hpc.social/results/hpc-social-design-survey.html](https://survey.hpc.social/results/hpc-social-design-survey.html) to see the range of responses. Let me know if you think we need to do more in this area.

**Reply by sramkrishna** [reply]
What do you think of using this as the basis of the code of conduct for mastodon? I think though that blogs are also covered under this. Or you need to put some kind of disclaimer for the blogs that the opinions are solely the authors/organization and not an endorsement.
https://www.contributor-covenant.org/version/2/1/code_of_conduct/code_of_conduct.md

**Reply by alansill** [reply]
The code of conduct for our Mastodon instance is pretty well covered by the existing requirements from the hosting provider and our rules that are already linked on the instance and in the existing repository. I think the primary need for code of conduct provisions would be on the other services (e.g., Slack, Discord, GitHub Discussions) but if we can find a forulation that can cover them all, that would be okay with me.

**Comment by sramkrishna** [comment]
I would say the GitHub discussions here as well.
